File: languages/java/src/main/java/com/bloock/sdk/entity/encryption/DecrypterArgs.java
package com.bloock.sdk.entity.encryption;

import com.bloock.sdk.entity.key.LocalKey;
import com.bloock.sdk.entity.key.ManagedKey;

public class DecrypterArgs {
  LocalKey localKey;
  ManagedKey managedKey;

  public DecrypterArgs(LocalKey localKey) {
    this.localKey = localKey;
  }

  public DecrypterArgs(ManagedKey managedKey) {
    this.managedKey = managedKey;
  }
}

File: languages/js/src/entity/encryption/decrypter_args.ts
import { LocalKey, ManagedKey } from "../key";

export class DecrypterArgs {
  localKey?: LocalKey;
  managedKey?: ManagedKey;

  constructor(key: LocalKey | ManagedKey | string) {
    if (key instanceof LocalKey) {
      this.localKey = key;
    } else if (key instanceof ManagedKey) {
      this.managedKey = key;
    } else {
      this.localKey = new LocalKey("", key);
    }
  }
}
